src/replication.c

```diff
@@ -399,13 +355,15 @@ void clear_replica_cmd(spec_t *spec, replica_t *replica, rcmd_t *rep_cmd) {
 int
 remove_replica_from_list(spec_t *spec, int iofd) {
 	replica_t *replica;
-	int ios_aborted = 0;
+	int ios_aborted = 0, rc;
 	rcmd_t *rep_cmd = NULL;
+	struct epoll_event event;
 	MTX_LOCK(&spec->rq_mtx);
 	TAILQ_FOREACH(replica, &spec->rq, r_next) {
 		if(iofd == replica->iofd) {
 			REPLICA_LOG("REMOVE REPLICA FROM LIST\n");
 			MTX_LOCK(&replica->r_mtx);
+			replica->state = REPLICA_ERRORED;
```

Comment:
here also, order need to be maintained.
First, update_volstate after updating the states and healthy/degraded replicas counters (updating counters can be part of cstor#137), then, remove from linked list and then signaling to stop threads and then closing fds.